### ISSUE TITLE
missing API method closewrite for Pkg.UnstableIO

### ISSUE THREAD

**Issue by vtjnash** [JuliaLang/Pkg.jl#3722]
The UnstableIO type is missing the `closewrite` method which causes it to throw this error (which gets usually swallowed by precompile, but here I made it visible while checking for API mistakes)
```
    nested task error: MethodError: no method matching closewrite(::Pkg.UnstableIO)
    
    Closest candidates are:
      closewrite(!Matched::Base.BufferStream)
       @ Base stream.jl:1487
      closewrite(!Matched::Base.LibuvStream)
       @ Base stream.jl:437
      closewrite(!Matched::Base.GenericIOBuffer)
       @ Base iobuffer.jl:350
      ...
    
    Stacktrace:
     [1] (::Base.var"#848#849"{Pkg.UnstableIO, Bool, Base.PipeEndpoint, Pkg.UnstableIO, Base.PipeEndpoint})()
       @ Base ./process.jl:318
in expression starting at /home/vtjnash/julia/usr/share/julia/stdlib/v1.11/Pkg/src/precompile.jl:6
in expression starting at /home/vtjnash/julia/usr/share/julia/stdlib/v1.11/Pkg/src/Pkg.jl:3
in expression starting at stdin:3
```

**Comment by vtjnash** [comment]
At that point of precompile, it just usually happens in Task that is missing an `errormonitor` so you miss it

**Comment by vtjnash** [comment]
It always happens, we just currently usually swallow the error tetx

**Comment by vtjnash** [comment]
It also probably means there is a data race and concurrency violation currently, as this error means Pkg is reading the stdout result before it is fully written on another thread (which I hope to fix on master though for v1.11)
